okhttp-apache/src/main/java/com/squareup/okhttp/apache/OkApacheClient.java

```diff
@@ -67,6 +69,8 @@ private static Request transformRequest(HttpRequest request) {
         if (encoding != null) {
           builder.header(encoding.getName(), encoding.getValue());
         }
+      } else {
```

Comment:
Apache HttpClient is able to send POST/PUT request without entity body.
For example following code works properly.

```java
HttpPut put = new HttpPut("URL");
HttpClient client = new DefaultHttpClient();
client.execute(put);
```

But OkApacheClient(master blanch) will throw IllegalArgumentException.
java.lang.IllegalArgumentException: method PUT must have a request body.

```java
HttpPut put = new HttpPut("URL");
OkApacheClient client = new OkApacheClient();
client.execute(put);
```